oi/OIDebugger.cpp

```diff
@@ -2918,15 +2922,24 @@ std::optional<std::string> OIDebugger::generateCode(const irequest& req) {
     return std::nullopt;
   }
 
-  if (auto sourcePath = cache.getPath(req, OICache::Entity::Source)) {
-    std::ofstream(*sourcePath) << code;
-  }
-
   typeInfos.emplace(
       req,
       std::make_tuple(RootInfo{rootInfo.varName, codegen->getRootType()},
                       codegen->getTypeHierarchy(), codegen->getPaddingInfo()));
 
+  if (useTypeGraph) {
```

Comment:
As this is only in one place, it would probably be cleaner to leave the signature alone and check `generatorConfig.features[Feature::TypeGraph]` explicitly.